control/controlclient/direct.go

```diff
@@ -72,6 +73,9 @@ type Direct struct {
 	mu             sync.Mutex        // mutex guards the following fields
 	serverKey      key.MachinePublic // original ("legacy") nacl crypto_box-based public key
 	serverNoiseKey key.MachinePublic
+	noiseClient    *noiseClient
+
+	sfGroup singleflight.Group
```

Comment:
say what it does/guards, how it's used, something. Otherwise it's just:

`integer int`

Reply:
Done